### PR TITLE
Processing data in Models

### PR DESCRIPTION
**Pull request by desorgui**
The models include some additional methods.
-  Users
    - A method that returns the 3 most recent posts for a given user.
-  Posts
    - A method that updates the posts counter for a user.
    - A method which returns the 5 most recent comments for a given post.
-  Comments
    - A method that updates the comments counter for a post.
-  Likes
    - A method that updates the likes counter for a post.

### CODE REVIEW

File: app/models/post.rb [hunk 1]
```diff
@@ -0,0 +1,22 @@
+class Post < ApplicationRecord
+  belongs_to :author, class_name: 'User', foreign_key: :author_id
```
Comment (rloterh) [inline app/models/post.rb]: - [x] Foreign key already declared in the _User_ model, and hence no need to declare it again in a referenced model. Simply do something like:
```ruby
# app/models/post.rb
belongs_to :author, class_name: 'User'

File: app/models/like.rb [hunk 2]
```diff
@@ -0,0 +1,12 @@
+class Like < ApplicationRecord
+  belongs_to :author, class_name: 'User', foreign_key: :author_id
```
Comment (rloterh) [inline app/models/like.rb]: - [x] Foreign key already declared in the _User_ model, and hence no need to declare it again in a referenced model. Simply do:
```ruby
# app/models/like.rb
belongs_to :author, class_name: 'User'

File: app/models/comment.rb [hunk 3]
```diff
@@ -0,0 +1,12 @@
+class Comment < ApplicationRecord
+  belongs_to :post
+  belongs_to :author, class_name: 'User', foreign_key: :author_id
```
Comment (rloterh) [inline app/models/comment.rb]: - [x] Foreign key already declared in the _User_ model, and hence no need to declare it again in a referenced model. Simply do:
```ruby
# app/models/comment.rb
belongs_to :author, class_name: 'User'

**Review by rloterh — CHANGES_REQUESTED** [review]
Hi Desor, 

You did extremely great on this project! 👏 👏 
There are however a few issues that you still need to work on to go to the next project but you are almost there! 🚦 🚦 🚦  

## Highlights 👍  
- Good job so far implementing model associations
- Properly implemented `after_save` method


## Required Changes ♻️
_Kindly check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

**You can also consider:**
- MIT license link in your ReadMe is broken, kindly endeavor to fix it.
Implementing the above change(s) will make your ReadMe appear more professional.

<br>
Cheers and Happy coding!👏 💻 🍷

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/models/comment.rb (hunk 3 above)
Comment (rloterh) [reply app/models/comment.rb]: well resolved! ✔️

File: app/models/like.rb (hunk 2 above)
Comment (rloterh) [reply app/models/like.rb]: well resolved! ✔️

File: app/models/post.rb (hunk 1 above)
Comment (rloterh) [reply app/models/post.rb]: well resolved! ✔️

**Comment by desorgui** [comment]
Thank you @rloterh

**Review by rloterh — APPROVED** [review]
## Status: Approved :heavy_check_mark:

Hi Desor,  

Your project is complete, great job 🥇! There is nothing else to say other than... it's time to merge it 🍾🚢 :shipit: . Congratulations!  🎉💯🌟 

## Highlights 👍
- Models associations correctly implemented

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

**You can also consider:**
- MIT license link in your ReadMe is broken, kindly endeavor to fix it.
Implementing the above change(s) will make your ReadMe appear more professional.
<br>
Keep soaring higher! 💻 🍷 🚀 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**